From a67a0bcebcb46078f51eb687585a8c787ecedb0f Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Mon, 28 May 2018 12:51:46 +0100
Subject: [PATCH 2/2] state: Do not track "<" mtime, make sure mtime is just
 different

Due to how package history reverts work, users may deliberately downgrade
their history to older package sets, or indeed we may forcibly sideload an
older package that is reinstalled. These files have older mtimes than the
state tracker, and should still be recorded again to ensure we can take the
system in either direction safely.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/state.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/state.c b/src/state.c
index 66bc4f5..a1b1e16 100644
--- a/src/state.c
+++ b/src/state.c
@@ -306,8 +306,8 @@ bool usc_state_tracker_needs_update(UscStateTracker *self, const char *path, boo
                 return true;
         }
 
-        /* If our record mtime is older than the current mtime, update it */
-        if (entry->mtime < mtime) {
+        /* If our record mtime is not the same as the current mtime, update it */
+        if (entry->mtime != mtime) {
                 return true;
         }
 
-- 
2.17.0

